app/src/main/java/org/p2p/wallet/history/interactor/TransactionDetailsMapper.kt

```diff
@@ -0,0 +1,258 @@
+            when (it) {
+                is CreateAccountDetails -> {
+                    CreateAccountTransactionEntity(
+                        identifiers = identifiers,
+                        commonInformation = commonInformation,
+                        fee = it.fee
+                    )
+                }
+                is CloseAccountDetails -> {
+                    CloseAccountTransactionEntity(
+                        identifiers = identifiers,
+                        commonInformation = commonInformation,
+                        account = it.account,
+                        mint = it.mint
+                    )
+                }
+                is BurnOrMintDetails -> {
+                    RenBtcBurnOrMintTransactionEntity(
+                        identifiers = identifiers,
+                        commonInformation = commonInformation,
+                        account = it.account,
+                        authority = it.authority,
+                        amount = it.amount,
+                        decimals = it.decimals,
+                        fee = it.fee
+                    )
+                }
+                is SwapDetails -> {
+                    SwapTransactionEntity(
+                        identifiers = identifiers,
+                        commonInformation = commonInformation,
+                        aEntity = SwapAEntity(
+                            mint = it.mintA,
+                            amount = it.amountA,
+                            source = it.source.toBase58Instance(),
+                            alternateSource = it.alternateSource.toBase58Instance()
+                        ),
+                        bEntity = SwapBEntity(
+                            mint = it.mintB,
+                            amount = it.amountB,
+                            destination = it.destination.toBase58Instance(),
+                            alternateDestination = it.alternateDestination.toBase58Instance()
+                        ),
+                        fee = it.fee
+                    )
+                }
+                is TransferDetails -> {
+                    TransferTransactionEntity(
+                        identifiers = identifiers,
+                        commonInformation = commonInformation,
+                        source = it.source.toBase58Instance(),
+                        destination = it.destination.toBase58Instance(),
+                        authority = it.authority.orEmpty(),
+                        mint = it.mint,
+                        amount = it.amount,
+                        decimals = it.decimals,
+                        fee = it.fee
+                    )
+                }
+                is UnknownDetails -> {
+                    UnknownTransactionEntity(
+                        identifiers = identifiers,
+                        commonInformation = commonInformation,
+                        rawData = Gson().toJson(it.info),
```

Comment:
Let's inject gson class here to avoid creating `Gson` object every time

Reply:
yep, will do